apps/api/src/server/applications/examination-timetable-items/examination-timetable-items.controller.js

```diff
@@ -33,5 +35,34 @@ export const createExaminationTimetableItem = async (_request, response) => {
 	const { body } = _request;
 	const examinationTimetableItem = await exminationTimetableItemsRepository.create(body);
 
+	// find the examination folder id to create subfolder.
+	let examinationFolder = await folderRepository.getFolderByNameAndCaseId(
+		body.caseId,
+		'Examination'
+	);
+
+	if (!examinationFolder) {
+		examinationFolder = await folderRepository.createFolder({
+			displayNameEn: 'Examination',
+			caseId: body.caseId,
+			parentFolderId: null,
+			displayOrder: 100
+		});
+	}
+
+	// in some case we don't have date, use mandatory field end date in that case.
+	const date = examinationTimetableItem.date || examinationTimetableItem.endDate;
+	// Create sub folder for the examination timetable item.
+	const folderName = `${format(new Date(date), 'dd MMM yyyy')} - ${examinationTimetableItem.name}`;
+
+	const folder = {
+		displayNameEn: folderName,
+		caseId: body.caseId,
+		parentFolderId: examinationFolder?.id || null,
+		displayOrder: 100
```

Comment:
I didn't know we had a display order, we can potentially use this instead of the ISO date!

Reply:
Ah yeah I've just noticed - we're creating timetable items separately here. We could use the unix timestamp of the date?